enhancements/oc/must-gather-collection-mode.md

```diff
@@ -0,0 +1,207 @@
+```
+
+If it is decided that the logs of daemonsets running on worker nodes are essential, additionally this command is an example;
+
+```shell
+ oc adm inspect namespace openshift-multus --node-selector='node-role.kubernetes.io/worker'
+```
+
+### API Extensions
+
+There is no API related change.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+No impact
+
+#### Standalone Clusters
+
+No impact
+
+#### Single-node Deployments or MicroShift
+
+No impact
+
+### Implementation Details/Notes/Constraints
+
+`--collection-mode` flag will be invisible and that would be difficult to find and use it without any prior knowledge.
+People may complain about the excessive duration but in reality, there is a way we just don't to expose them.
+
+### Risks and Mitigations
+
+Some logs are skipped in _limited_ mode, even though they are essential. That causes the less usability of limited mode.
+Because it is an additional requirement and back and forth to collect more data with a separate command.
+
+### Drawbacks
+
+This brings about an inevitable but slightly maintenance burden as it introduces new flags and environment variables.
+We have to skip the collection of some logs due to the constraints we have and the drawback is some logs are not essential 
+for some clusters but contrarily these logs could be very essential for some clusters. This proposal eliminates 
+some logs which may end up that limited mode can't be usable for some clusters.
+
+## Open Questions [optional]
+
+None
+
+## Test Plan
+
+We have several pinpoints in our CI testing the must-gather's default behavior. That can assure that this change does not
+break it.
+
+## Graduation Criteria
+
+### Dev Preview -> Tech Preview
+
+In this case, we can merge Dev Preview and Tech Preview and we can have this;
+
+* `--collection-mode` flag will be hidden in must-gather, but it is triggerable if you know the flag
+* `--node-selector` flag will be visible but marked as experimental
+
+### Tech Preview -> GA
+
+* After the adoption of `--collection-mode` flag by other must-gather images, we can mark this flag as visible for general use
+* `--node-selector` will be marked as GA by removing the experimental
+
+### Removing a deprecated feature
+
+None
+
+## Upgrade / Downgrade Strategy
+
+None
+
+## Version Skew Strategy
+
+* oc that is used to invoke must-gather is old, regardless of must-gather is new or old; no issue as the environment variable will not be set.
+* oc that is used to invoke must-gather is new, must-gather script is old; no issue, environment variable will be ignored
+* oc that is used to invoke must-gather is new, must-gather script is new, oc in must-gather script is old; since the old oc in must-gather does not know `--node-selector`, it will fail
+* oc that is used to invoke must-gather is new, must-gather script is old, oc in must-gather script is new; since must-gather script does not pass `--node-selector`, no issue
```

Comment:
Neither of this situations can happen and if they did, it would be a bug:

In OCP, we deliver full release payloads and the must-gather image is one of the images within that payload. So we can either have:
- A must-gather image with an `oc` that understands node-selector in inspect and the `gather` script that leverages that for collection profiles
- A must-gather image with an older `oc` that doesn't have node-selector in inspect but also doesn't have a `gather` script trying to use that.

If we had an incompatible combination of oc and gather script in a must-gather image, that would be a bug in that image, because we should have never built such an image. So I don't think we need to document this in the strategy, maybe just somewhere else warn that we need to be careful with this while doing backports.

Reply:
The same reasoning holds for `--since` and `--since-time` flags. They translate into pod envs. It is not assumed any MG image will interpret it properly. So unless a MG image of a given version already adds support for all the bug reports will be interpreted as feature requests and forwarded to owning teams.

Reply:
What you are saying in this comment is not what is referred in the points I signaled but in the previous point:
```
* oc that is used to invoke must-gather is new, must-gather script is old; no issue, environment variable will be ignored
```
In such cases, as the point already explains, the environment variable is just ignored and it's just fine.

In these two points that I signaled it is suggested is that the same must-gather image has a script that understands the new flags and an oc client which doesn't or vice-versa. That's an inconsistency in the must-gather image.

Either the must-gather image fully adopts the flag, at both its oc client and its script, or it doesn't. Adopting the flags is indeed a new feature. Half-adopting the flags as you suggest in either of these points, is a bug.